File: src/fe/src/models/WorklistFromBioHubItem.ts
import { WorklistFromBioHubStatus } from "./enums/WorklistFromBioHubStatus";
import { DocumentFileType } from "./enums/DocumentFileType";
import { RoleType } from "@/models/enums/RoleType";
import { WorklistFillingOption } from "@/models/enums/WorklistFillingOption";
import { ShipmentDocument } from "@/models/ShipmentDocument";
import { WorklistFromBioHubItemFeedback } from "@/models/WorklistFromBioHubItemFeedback";
import { WorklistFromBioHubItemMaterial } from "./WorklistFromBioHubItemMaterial";
import { WorklistFromBioHubItemBiosafetyChecklistOfSMTA2 } from "./WorklistFromBioHubItemBiosafetyChecklistOfSMTA2";
import { WorklistFromBioHubItemAnnex2OfSMTA2Condition } from "./WorklistFromBioHubItemAnnex2OfSMTA2Condition";
import { WorklistFromBioHubItemBiosafetyChecklistThreadComment } from "./WorklistFromBioHubItemBiosafetyChecklistThreadComment";
import { SMTAApprovalStatus } from "./enums/SMTAApprovalStatus";
import { WorklistItemUser } from "./WorklistItemUser";
import { BookingFormOfSMTA } from "./BookingFormOfSMTA";

export enum ShipmentDocumentOperationType {
  Add = 0,
  Update = 1,
  Delete = 2,
}

export interface WorklistFromBioHubItem {
  Id: string;
  CurrentStatus: WorklistFromBioHubStatus;
  CurrentStatusName: string;
  PreviousStatus: WorklistFromBioHubStatus;
  WorklistItemTitle: string;
  LastSubmissionApproved: boolean;
  LaboratoryName: string;
  LaboratoryAddress: string;
  LaboratoryAbbreviation: string;
  LaboratoryCountry: string;
  BioHubFacilityName: string;
  BioHubFacilityAddress: string;
  BioHubFacilityCountry: string;
  OperationDate: Date;
  UserName: string;
  Comment: string;
  SMTA2DocumentId: string;
  SMTA2DocumentName: string;
  LaboratoryId: string;
  BioHubFacilityId: string;
  OriginalDocumentTemplateSMTA2DocumentId: string;
  DocumentTemplateFileType: DocumentFileType;
  HistoryDto: boolean;
  UserRoleName: string;
  UserRoleTypeName: string;
  UserRoleType: RoleType;
  OriginalDocumentTemplateAnnex2OfSMTA2DocumentId: string;
  Annex2OfSMTA2DocumentId: string;
  Annex2OfSMTA2DocumentName: string;
  Annex2OfSMTA2SignatureId: string;
  Annex2OfSMTA2SignatureString: string;
  Annex2Comment: string;
  Annex2TermsAndConditions: boolean;
  Annex2FillingOption: WorklistFillingOption;
  WorklistFromBioHubItemMaterials: Array<WorklistFromBioHubItemMaterial>;
  LaboratoryFocalPoints: Array<WorklistItemUser>;
  WorklistFromBioHubItemAnnex2OfSMTA2Conditions: Array<WorklistFromBioHubItemAnnex2OfSMTA2Condition>;
  WorklistFromBioHubItemBiosafetyChecklistOfSMTA2s: Array<WorklistFromBioHubItemBiosafetyChecklistOfSMTA2>;
  Annex2ApprovalFlag: boolean;
  Annex2ApprovalComment: string;
  IsSaveDraft: boolean;
  BookingForms: Array<BookingFormOfSMTA>;
  BookingFormFillingOption: WorklistFillingOption;
  OriginalDocumentTemplateBookingFormOfSMTA2DocumentId: string;
  BookingFormOfSMTA2DocumentId: string;
  BookingFormOfSMTA2DocumentName: string;
  BookingFormOfSMTA2SignatureId: string;
  BookingFormOfSMTA2SignatureString: string;
  BookingFormApprovalFlag: boolean;
  BookingFormApprovalComment: string;
  BiosafetyChecklistFillingOption: WorklistFillingOption;
  OriginalDocumentTemplateBiosafetyChecklistOfSMTA2DocumentId: string;
  BiosafetyChecklistOfSMTA2DocumentId: string;
  BiosafetyChecklistOfSMTA2DocumentName: string;
  BiosafetyChecklistOfSMTA2SignatureId: string;
  BiosafetyChecklistOfSMTA2SignatureString: string;
  BiosafetyChecklistOfSMTA2ApprovalFlag: boolean;
  BiosafetyChecklistOfSMTA2ApprovalComment: string;
  LastOperationUserFirstName: string;
  LastOperationUserLastName: string;
  LastOperationUserBusinessPhone: string;
  LastOperationUserMobilePhone: string;
  LastOperationUserEmail: string;
  LastOperationUserJobTitle: string;
  ShipmentDocumentOperationType: ShipmentDocumentOperationType;
  ShipmentDocumentId: string;
  ShipmentDocumentNewName: string;
  BHFShipmentDocuments: Array<ShipmentDocument>;
  QEShipmentDocuments: Array<ShipmentDocument>;
  PackagingListDocumentTemplateId: string;
  PackagingListDocumentTemplateName: string;
  NonCommercialInvoiceCatADocumentTemplateId: string;
  NonCommercialInvoiceCatADocumentTemplateName: string;
  NonCommercialInvoiceCatBDocumentTemplateId: string;
  NonCommercialInvoiceCatBDocumentTemplateName: string;
  WaitForArrivalConditionCheckApprovalComment: string;
  WaitForArrivalConditionCheckApprovalFlag: boolean;
  NewFeedback: string;
  Feedbacks: Array<WorklistFromBioHubItemFeedback>;
  WHODocumentRegistrationNumber: string;

  NewBiosafetyChecklistThreadComment: string;
  BiosafetyChecklistThreadComments: Array<WorklistFromBioHubItemBiosafetyChecklistThreadComment>;

  PreviousOperationDate: Date;
  PreviousUserId: string;

  PreviousActionBy: string;
  NextActionBy: string;
  Institution: string;
  LaboratoryCountryName: string;

  CurrentDownloadSMTA2DocumentId: string;
  SMTA2ApprovalStatus: SMTAApprovalStatus;
  SMTA2ApprovalDate: Date | null;
  CurrentDownloadSMTA2DocumentName: string;

  IsPast: boolean;
  AssignedOperationDate: Date | null;

  //# 54317
  Annex2OfSMTA2SignatureText: string;
  BookingFormOfSMTA2SignatureText: string;
  BiosafetyChecklistOfSMTA2SignatureText: string;
  ///////
}

File: src/be/src/WHO.BioHub.Models/Models/WorklistFromBioHubItem.cs
using WHO.BioHub.Models.Models.Abstractions;
using WHO.BioHub.Shared.Enums;

namespace WHO.BioHub.Models.Models;

public class WorklistFromBioHubItem : EntityBase
{
    public WorklistFromBioHubStatus Status { get; set; }
    public WorklistFromBioHubStatus PreviousStatus { get; set; }
    public string WorklistItemTitle { get; set; }
    public DateTime? OperationDate { get; set; }
    public bool Completed { get; set; }
    public Guid? RequestInitiationToLaboratoryId { get; set; }
    public virtual Laboratory RequestInitiationToLaboratory { get; set; }
    public Guid? RequestInitiationFromBioHubFacilityId { get; set; }
    public virtual BioHubFacility RequestInitiationFromBioHubFacility { get; set; }
    public Guid? LastOperationUserId { get; set; }
    public virtual User LastOperationUser { get; set; }
    public bool? LastSubmissionApproved { get; set; }
    public string Comment { get; set; }
    public virtual ICollection<WorklistFromBioHubHistoryItem> WorklistFromBioHubHistoryItems { get; set; }
    public virtual ICollection<WorklistFromBioHubItemDocument> WorklistFromBioHubItemDocuments { get; set; }
    public string Annex2Comment { get; set; }
    public bool? Annex2TermsAndConditions { get; set; }
    public FillingOption? Annex2FillingOption { get; set; }
    public bool? Annex2ApprovalFlag { get; set; }
    public string Annex2ApprovalComment { get; set; }
    public DateTime? Annex2OfSMTA2ApprovalDate { get; set; }
    public Guid? OriginalAnnex2OfSMTA2DocumentTemplateId { get; set; }

    public FillingOption? BiosafetyChecklistFillingOption { get; set; }
    public bool? BiosafetyChecklistApprovalFlag { get; set; }
    public string BiosafetyChecklistApprovalComment { get; set; }
    public DateTime? BiosafetyChecklistApprovalDate { get; set; }
    public Guid? OriginalBiosafetyChecklistDocumentTemplateId { get; set; }
    public virtual ICollection<BookingForm> BookingForms { get; set; }
    public FillingOption? BookingFormFillingOption { get; set; }
    public bool? BookingFormApprovalFlag { get; set; }
    public string BookingFormApprovalComment { get; set; }
    public DateTime? BookingFormOfSMTA2ApprovalDate { get; set; }
    public Guid? OriginalBookingFormOfSMTA2DocumentTemplateId { get; set; }
    public bool? WaitForArrivalConditionCheckApprovalFlag { get; set; }
    public string WaitForArrivalConditionCheckApprovalComment { get; set; }
    public virtual ICollection<WorklistFromBioHubItemFeedback> WorklistFromBioHubItemFeedbacks { get; set; }
    public string WHODocumentRegistrationNumber { get; set; }
    public string ReferenceNumber { get; set; }
    public virtual ICollection<Shipment> Shipments { get; set; }
    public Guid? ReferenceId { get; set; }
    public virtual ICollection<WorklistFromBioHubItemAnnex2OfSMTA2Condition> WorklistFromBioHubItemAnnex2OfSMTA2Conditions { get; set; }
    public virtual ICollection<WorklistFromBioHubItemBiosafetyChecklistOfSMTA2> WorklistFromBioHubItemBiosafetyChecklistOfSMTA2s { get; set; }
    public virtual ICollection<WorklistFromBioHubItemMaterial> WorklistFromBioHubItemMaterials { get; set; }
    public virtual ICollection<WorklistFromBioHubItemLaboratoryFocalPoint> WorklistFromBioHubItemLaboratoryFocalPoints { get; set; }

    public virtual ICollection<WorklistFromBioHubItemBiosafetyChecklistThreadComment> WorklistFromBioHubItemBiosafetyChecklistOfSMTA2Comments { get; set; }
    public string SavedBiosafetyChecklistThreadComment { get; set; }
    public bool? IsPast { get; set; }

    //# 54317
    public string Annex2OfSMTA2SignatureText { get; set; }
    public string BookingFormOfSMTA2SignatureText { get; set; }
    public string BiosafetyChecklistOfSMTA2SignatureText { get; set; }
    ///////
}

